src/Riak/Command/Search/Response.php

```diff
@@ -35,7 +35,7 @@ public function __construct($statusCode, $headers = [], $body = '')
         parent::__construct($statusCode, $headers, $body);
 
         // make sure body is not only whitespace
-        if (trim($body)) {
+        if (trim($body) && $this->statusCode != 400) {
```

Comment:
Are there other error status codes you want to exclude here?  Or simply say `$this->statusCode == 200`?